src/tests/ftest/io/EightServers_mpiio.py

```diff
@@ -0,0 +1,168 @@
+
+        # setup the DAOS python API
+        self.Context = DaosContext(build_paths['PREFIX'] + '/lib/')
+
+        self.hostlist_servers = self.params.get("test_servers", '/run/hosts/test_machines/*')
+        self.hostfile_servers = WriteHostFile.WriteHostFile(self.hostlist_servers, self.workdir)
+        print("Host file servers is: {}".format(self.hostfile_servers))
+
+        self.hostlist_clients = self.params.get("test_clients", '/run/hosts/test_machines/*')
+        self.np = self.params.get("np", '/run/ior/client_processes/*')
+        self.hostfile_clients = WriteHostFile.WriteHostFile(self.hostlist_clients, self.workdir)
+        print("Host file clients is: {}".format(self.hostfile_clients))
+
+        ServerUtils.runServer(self.hostfile_servers, self.server_group, self.basepath)
+
+    def tearDown(self):
+        try:
+            if self.pool is not None and self.pool.attached:
+                self.pool.destroy(1)
+        finally:
+            ServerUtils.stopServer(hosts=self.hostlist_servers)
+
+    def executable(self, iorflags=None):
+        """
+        Executable function to run ior for ssf and fpp
+        """
+
+        # parameters used in pool create
+        createmode = self.params.get("mode", '/run/pool/createmode/*/')
+        createuid = os.geteuid()
+        creategid = os.getegid()
+        createsetid = self.params.get("setname", '/run/pool/createset/')
+        createscm_size = self.params.get("scm_size", '/run/pool/createsize/')
+        createnvme_size = self.params.get("nvme_size", '/run/pool/createsize/')
+        createsvc = self.params.get("svcn", '/run/pool/createsvc/')
+        iteration = self.params.get("iter", '/run/ior/iteration/')
+        block_size = self.params.get("b", '/run/ior/transfersize_blocksize/*/')
+        transfer_size = self.params.get("t", '/run/ior/transfersize_blocksize/*/')
+        expected_result = 'PASS'
+
+        try:
+            # Do not need slots in hostfile
+            with open(self.hostfile_clients) as f:
+                newText = f.read().replace('slots=1', '')
+
+            with open(self.hostfile_clients, "w") as f:
+                f.write(newText)
+
+            # initialize MpioUtils
+            self.mpio = MpioUtils()
+            if self.mpio.mpich_installed(self.hostlist_clients) is False:
+                self.fail("Exiting Test: Mpich not installed")
+
+            # initialize a python pool object then create the underlying
+            # daos storage
+            self.pool = DaosPool(self.Context)
+            self.pool.create(createmode, createuid, creategid,
+                             createscm_size, createsetid, None, None, createsvc,
+                             createnvme_size)
+
+            pool_uuid = self.pool.get_uuid_str()
+            svc_list = ""
+            for i in range(createsvc):
+                svc_list += str(int(self.pool.svc.rl_ranks[i])) + ":"
+            svc_list = svc_list[:-1]
```

Comment:
So why do we create this string if we only want the last thing that was appended?  Isn't this the same as svc_list = str(int(self.pool.svc.rl_ranks[createsvc-1]))

Reply:
This is not using last thing that was appended. This will remove the last thing that was appended.
Example: if we use svcl=3 we want svc_list=1:2:3 but with svc_list += str(int(self.pool.svc.rl_ranks[i])) + ":" we will get 1:2:3: and hence we want to remove that final ":"

Reply:
yeah, okay.  Nothing before the : means beginning of the list and -1 means except the last element.  I think in C++ or Java and just transmute it to python when I type so I never use all that cool stuff.